code/game/objects/items/devices/silent_alarm.dm

```diff
@@ -0,0 +1,49 @@
+/obj/machinery/button/silent_alarm
+	name = "Silent alarm"
+	desc = "A small button that silently notifies security when pressed"
+	silent = TRUE
+	device_type = /obj/item/assembly/silent_alarm
+
+/obj/item/assembly/silent_alarm
+	name = "Silent alarm electronics"
+	desc = "The internal electronics of a silent alarm button"
+	icon_state = "control"
+	COOLDOWN_DECLARE(announce_cooldown)
+	assembly_flags = ASSEMBLY_SILENCE_BUTTON | ASSEMBLY_NO_DUPLICATES
+
+/obj/item/assembly/silent_alarm/activate(mob/user)
+	if(is_within_radio_jammer_range(src))
+		COOLDOWN_START(src, announce_cooldown, 1.5 MINUTES)
+		return
+	var/location = get_area_name(user)
+	if(!location)
+		location = get_area_name(src)
```

Comment:
Surely this should just use `get_area_name(src)` otherwise people with telekinesis would make it report from somewhere else